SortingAlgorithm/Sorting Console Application/Sorting Console Application.cpp

```diff
@@ -0,0 +1,86 @@
+// Sorting Console Application.cpp : main project file.
+
+#include "stdafx.h"
+#include<iostream>
+#include<fstream>
+#include <vcclr.h>
+#include <string>
+#include "FileInputOutput.h"
+
+using namespace System;
+using namespace SortingAlgorithm;
+
+
+class SortingFromFile
+{
+	gcroot<Sort^> sorting;
+	FileInputOutput* fileInputOutput;
+	int *arr, size;
+
+public:
+	SortingFromFile(String^ inFileName, String^ outFileName, Sort^ sort)
+	{
+		this->fileInputOutput = new FileInputOutput(inFileName, outFileName);
+		this->sorting = sort;
+		this->arr = new int[1000];
+	}
+	
+	~SortingFromFile()
+	{
+		delete fileInputOutput;
+		delete[] arr;
+		std::cout << "Deleted native pointer upon calling this destructor" << std::endl;
+	}
+
+	void readFile()
+	{
+		fileInputOutput->readFile(arr);
+
+		// Size of the array
+		size = fileInputOutput->getNumberOfElements();
+	}
+
+	void writeFile()
+	{
+		fileInputOutput->writeFile(arr);
+	}
+
+	void printArray()
+	{
+		for (int i = 0; i < size; i++)
+		{
+			std::cout << arr[i] << std::endl;
+		}
+	}
+
+	void sort()
+	{
+		std::cout << "Before sort:" << std::endl;
```

Comment:
Here we are using arr to show unsorted and sorted list to verify our algorithm. But can't we read the input and output files to show the changes. and update printArray accordingly.

Reply:
Yes, we can. But that would result in some duplicate work and also File IO is much slower. As we have already loaded the numbers by reading from a file, so we are printing the unsorted file from the array and after we are finished sorting this array, then before writing to file, we are again printing the numbers to console from the same array to show the changes this sorting algorithm has enforced.